packages/flutter/lib/src/material/search.dart

```diff
@@ -0,0 +1,476 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+
+import 'app_bar.dart';
+import 'colors.dart';
+import 'input_border.dart';
+import 'input_decorator.dart';
+import 'material_localizations.dart';
+import 'scaffold.dart';
+import 'text_field.dart';
+import 'theme.dart';
+
+/// Shows a full screen search overlay.
```

Comment:
"overlay" might be a little confusing vis the Overlay widget. The docs tend to refer to the result of showSearchOverlay as a page, show maybe showSearchPage etc.